avm/ptn/authorization/role-definition/main.bicep

```diff
@@ -0,0 +1,106 @@
+metadata name = 'avm/ptn/authorization/role-definition'
+metadata description = 'This module deploys a custom role definition to a Management Group.'
+metadata owner = '@jtracey93'
+
+targetScope = 'managementGroup'
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. The location of the telemetry deployment to be created. Default is location of deployment.')
+param location string = deployment().location
+
+@description('Required. Array of custom role definitions to create on the management group.')
+param roleDefinition roleDefinitionType
```

Comment:
This is usually something we ask contributors to split into dedicated parameters in alignment with the recommendations by the PG. One reason is, for example that you cannot define any defaults in the type and also cannot surface any to the user of the module.

Reply:
So split out all the nice UDT provided stuff into separate params? 

Can i still leave the type to expose to make it easier for folks to wrap an array of objects up with intellisense support?

@AlexanderSehr

Reply:
Have now altered this and will stick the UDT elsewhere in other modules whilst we await resource derived types